crates/bevy_reflect/src/struct_trait.rs

```diff
@@ -73,6 +73,10 @@ pub trait Struct: Reflect {
 
     /// Clones the struct into a [`DynamicStruct`].
     fn clone_dynamic(&self) -> DynamicStruct;
+
+    /// Returns the structs [`StructInfo`].
+    /// Will return `None` if [`TypeInfo`] is not available.
+    fn get_represented_kind_info(&self) -> Option<&StructInfo>;
```

Comment:
Could we make this return a `'static` reference?

Reply:
Ah I didn't realize that would be available. Changed